src/core/ActivityInfoTree.h

```diff
@@ -59,12 +61,17 @@ protected slots:
 private:
     // this is the full activity list, it never changes
     QList<ActivityInfo *> m_menuTreeFull;
+    QList<ActivityInfo *> m_menuTreeFulltemp;
     // represents the Menu view and can be filtered
     QList<ActivityInfo *> m_menuTree;
+    QList<ActivityInfo *> m_searched;
     ActivityInfo *m_rootMenu;
     ActivityInfo *m_currentActivity;
     static int menuTreeCount(QQmlListProperty<ActivityInfo> *property);
     static ActivityInfo *menuTreeAt(QQmlListProperty<ActivityInfo> *property, int index);
+    QObject *m_mainWindow;
+    int templength = 0;
+    QString searchedtext;
```

Comment:
for me, the templength is used to store the length of the text already typed. And if we add a letter in the textfield, we don't do the search on all the activities but only the ones in the temp tree (which is the one filled by the previous search).
Also, if templength is the length of the previous text, it can be a local variable to the function (set before searchedtext = text;)
With your solution we have 4 trees (the full one, the one we display and two for optimizing the search time).
My question is: is it slow to always recompute the tree at each character input (basically using same code as filterByTag/filterLockedActivities), or do we need to optimize it? In both case, I'm not sure it's needed to add the 2 other trees as we could do the same with only m_menuTree (same code you did, we only store the current search text as attribute and if its length is bigger than the next search we recompute all else we can search inside the already filtered m_menuTree).